sklearn/cluster/hierarchical.py

```diff
@@ -713,7 +713,7 @@ def fit(self, X, y=None):
         -------
         self
         """
-        X = check_array(X)
+        X = check_array(X, ensure_min_samples=2)
```

Comment:
This fails with 1 cluster as well, as we found out in this test
https://github.com/vighneshbirodkar/scikit-learn/blob/array_1d_fix/sklearn/utils/estimator_checks.py#L392